src/core/ext/client_channel/client_channel.c

```diff
@@ -1096,6 +1099,9 @@ void grpc_client_channel_finish_initialization(
       chand->exit_idle_when_lb_policy_arrives) {
     chand->started_resolving = true;
     GRPC_CHANNEL_STACK_REF(chand->owning_stack, "resolver");
+    if (chand->lb_policy != NULL) {
```

Comment:
Will this ever be true?  We know that the resolver was not previously set, due to the assertion on line 1095 above.  And since we haven't yet called grpc_resolver_next(), we know that on_resolver_result_changed() hasn't yet been called, and I believe that that's the only place where chand->lb_policy is being set.